tests/test_admin.py

```diff
@@ -241,3 +243,130 @@ def test_alias_content_manager_rendering_preview_add_url(self):
             add_aliascontent_url,
             response_content_decoded,
         )
+
+    def _create_alias_and_categories(self, category_name, alias_content_name=None):
+        if not alias_content_name:
+            alias_content_name = category_name
+        category = Category.objects.create(name=category_name)
+        alias = AliasModel.objects.create(category=category, position=0)
+        alias_content = AliasContent.objects.create(
+            alias=alias,
+            name=alias_content_name,
+            language="en"
+        )
+        return category, alias, alias_content
+
+    @skipUnless(is_versioning_enabled(), 'Test only relevant for versioning')
+    def test_category_field_ordering_versioned(self):
+        """
+        Related category can be ordered by name, both in ascending and descending order, with versioning
+        """
+        # Create a number of categories, aliases, and alias content to order
+        first_category, first_alias, first_alias_content = self._create_alias_and_categories("First Order Test Case")
+        # Previously lowercase and upper case would be sorted separately, test they are ordered together
+        first_category_lower, first_alias_lower, first_alias_content_lower = self._create_alias_and_categories(
+            "first order test case lower"
+        )
+        middle_category, middle_alias, middle_alias_content = self._create_alias_and_categories(
+            "Middle Order Test Case"
+        )
+        # Previously lowercase and upper case would be sorted separately, test they are ordered together
+        last_category_lower, last_alias_lower, last_alias_content_lower = self._create_alias_and_categories(
+            "z order test case lower"
+        )
+        last_category, last_alias, last_alias_content = self._create_alias_and_categories(
+            "Z Order Test Case Upper"
+        )
+        # Create the versions for each alias content
+        from djangocms_versioning.models import Version
+        Version.objects.create(content=first_alias_content, created_by=self.superuser)
+        Version.objects.create(content=first_alias_content_lower, created_by=self.superuser)
+        Version.objects.create(content=middle_alias_content, created_by=self.superuser)
+        Version.objects.create(content=last_alias_content_lower, created_by=self.superuser)
+        Version.objects.create(content=last_alias_content, created_by=self.superuser)
+
+        with self.login_user_context(self.superuser):
+            base_url = self.get_admin_url(AliasContent, "changelist")
+            # o=1 indicates ascending alphabetical order on list_displays second entry
+            base_url += "?o=1"
+            # en is the default language configured for the site
+            response = self.client.get(base_url)
+        soup = BeautifulSoup(response.content, "html.parser")
```

Comment:
You could have done queryset equals here too. Both solve the same issue so no change necessary.